### ISSUE TITLE
Error doing job: Task exception was never retrieved

### ISSUE THREAD

**Issue by robinhood-code** [jaydeethree/Home-Assistant-weatherdotcom#23]
I have been getting this error in HA log since I have updated to the latest version. 

Appreciate if anyone can advise the issue. 

Logger: homeassistant
Source: custom_components/weatherdotcom/sensor.py:126
Integration: Weather.com ([documentation](https://github.com/jaydeethree/Home-Assistant-weatherdotcom), [issues](https://github.com/jaydeethree/Home-Assistant-weatherdotcom/issues/))
First occurred: 12:04:27 AM (57 occurrences)
Last logged: 6:44:42 PM

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 243, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 399, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 182, in async_update_listeners
    update_callback()
  File "/config/custom_components/weatherdotcom/sensor.py", line 110, in _handle_coordinator_update
    self._sensor_data = _get_sensor_data(
                        ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/weatherdotcom/sensor.py", line 126, in _get_sensor_data
    return sensors[RESULTS_CURRENT][kind]
           ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
KeyError: 'latitude'

**Comment by nima-1102** [comment]
Same here

Integration Version: 1.1.4

Core: 2024.2.0
Supervisor: 2024.01.1
Operating System: 11.5
Frontend: 20240207.0


This error originated from a custom integration.

```
Logger: homeassistant
Source: custom_components/weatherdotcom/sensor.py:126
Integration: Weather.com (documentation, issues)
First occurred: 18:13:07 (1 occurrences)
Last logged: 18:13:07

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 256, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 412, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 183, in async_update_listeners
    update_callback()
  File "/config/custom_components/weatherdotcom/sensor.py", line 110, in _handle_coordinator_update
    self._sensor_data = _get_sensor_data(
                        ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/weatherdotcom/sensor.py", line 126, in _get_sensor_data
    return sensors[RESULTS_CURRENT][kind]
           ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
KeyError: 'latitude'

```

**Comment by jaydeethree** [comment]
Thanks for reporting this! This should be fixed by https://github.com/jaydeethree/Home-Assistant-weatherdotcom/commit/63f4713fbd07fad1d5cf66bd8a3b88aa17f7a33b and the fix is available in version 1.1.4.1.
